### ISSUE TITLE
Supports GEOIP2

### ISSUE THREAD

**Issue by yudao** [allinurl/goaccess#283]
Hi,

Now Maxmind has created a new database called GEOIP2 with more powerful and performance. The legacy one will be abandoned in future.

So it will be great to let goaccess support the legacy database but the new one too.

Regards,
Yudao

**Comment by allinurl** [comment]
Thanks for pointing this out. Do you know if they have a C API/library to access GEOIP2?

**Comment by yudao** [comment]
yes : https://github.com/maxmind/libmaxminddb

Sorry to not pointing it on my first post

**Comment by allinurl** [comment]
Awesome, I'll look into this. Thanks again!

**Comment by yudao** [comment]
No problem.
It will be great to not have a geoip compilation flag and read automatically form the library with the database path in the config file.
I'm using Mac OSX and homebrew and it's a pain to have recompile goaccess. 
But I guess there are maybe other restrictions for that.

**Comment by cicku** [comment]
Will patch soon.

**Comment by overnine** [comment]
Would it be possible to add support for the maxmind GEOIP2 City database perhaps also, along with adding the outputted coordinates? Rather than just country information?

**Comment by allinurl** [comment]
@overnine are the coordinates available on the free version of GeoIP2? Do you know which fields are available on the free version?

**Comment by overnine** [comment]
@allinurl I believe that they are indeed, it just contains much less records in the free version: https://www.maxmind.com/en/geoip-demo

**Comment by allinurl** [comment]
Thanks for the reminder, I'll definitely check on this. However, not sure if I can squeeze this in the upcoming release that I'm working right now (v1.0), otherwise, it will probably be out after v1.0 for sure.

**Comment by ix5** [comment]
Any update on this? Using a `.mmdb` from [GeoLite2](http://dev.maxmind.com/geoip/geoip2/geolite2/) is giving errors.

**Comment by allinurl** [comment]
Support for GeoIP2 has been added.

It now requires to configure goaccess as (GeoIP2):

    ./configure --enable-geoip=mmdb

or for legacy GeoIP

    ./configure --enable-geoip=legacy

Databases can be found [**here**](http://dev.maxmind.com/geoip/geoip2/geolite2/). And use `--geoip-database /path/GeoLite2-City.mmdb` to specify the database.

Please feel free to tested by building from [**upstream**](https://goaccess.io/download#build). It will be pushed out in the upcoming version.

**Comment by ix5** [comment]
Works great, thank you for implementing this, @allinurl.

For others: You will need `libmaxminddb-dev` on debian-based systems to compile.

**Comment by danbachar** [comment]
hey guys, 
I'm currently fighting this for the last couple of hours - I've been trying to run dockerized goaccess with geoip enabled, using GeoIP2 (mmdb, not legacy). I've been getting the same error in my docker container logs, regardless of whether I do this over `docker-compose` or `docker run`: `Unable to open GeoIP2 database: /goaccess/geoip/GeoLite2-City.mmdb`.
Worth mentioning is that the db file itself is located in a `geoip` volume which is attached to the container from the host, and I made sure that it is actually available inside the container, in the path specified in my goaccess conf file (with 755 permissions) 
The image is the latest, and I also inspected the Dockerfile to make sure it was running `configure` with `--enable-geoip=mmdb`, and it is. Not sure if this is relevant but the server running this is an Ubuntu 4.15.0-99-generic.

Thanks for the help!
Dan
